vendor/bat-native-ledger/src/bat/ledger/internal/report/report.cc

```diff
@@ -74,27 +69,20 @@ void Report::OnTransactions(
     ledger::TransactionReportInfoList transaction_report,
     const ledger::ActivityMonth month,
     const uint32_t year,
-    const std::string& monthly_report_string,
+    std::shared_ptr<ledger::MonthlyReportInfoPtr> shared_report,
     ledger::GetMonthlyReportCallback callback) {
-  auto monthly_report = braveledger_bind_util::FromStringToMonthlyReport(
-      monthly_report_string);
-
-  if (!monthly_report) {
+  if (!shared_report) {
     BLOG(0, "Could not parse monthly report");
     callback(ledger::Result::LEDGER_ERROR, nullptr);
     return;
   }
 
-  monthly_report->transactions = std::move(transaction_report);
-
-  const std::string callback_monthly_string =
-      braveledger_bind_util::FromMonthlyReportToString(
-          std::move(monthly_report));
+  (*shared_report)->transactions = std::move(transaction_report);
 
   auto contribution_callback = std::bind(&Report::OnContributions,
       this,
       _1,
-      callback_monthly_string,
+      std::make_shared<ledger::MonthlyReportInfoPtr>(std::move(*shared_report)),
```

Comment:
Can the shared pointer be passed directly into bind?

Reply:
fixed